cmd/snap-update-ns/update.go

```diff
@@ -39,3 +40,103 @@ type MountProfileUpdateContext interface {
 	// SaveCurrentProfile saves the mount profile that is currently applied.
 	SaveCurrentProfile(*osutil.MountProfile) error
 }
+
+func applySystemFstab(ctx MountProfileUpdateContext) error {
+	unlock, err := ctx.Lock()
+	if err != nil {
+		return err
+	}
+	defer unlock()
+
+	// Read the desired and current mount profiles. Note that missing files
+	// count as empty profiles so that we can gracefully handle a mount
+	// interface connection/disconnection.
+	desired, err := ctx.LoadDesiredProfile()
+	if err != nil {
+		return err
+	}
+	debugShowProfile(desired, "desired mount profile")
+
+	currentBefore, err := ctx.LoadCurrentProfile()
+	if err != nil {
+		return err
+	}
+	debugShowProfile(currentBefore, "current mount profile (before applying changes)")
+	// Synthesize mount changes that were applied before for the purpose of the tmpfs detector.
+	as := ctx.Assumptions()
+	for _, entry := range currentBefore.Entries {
+		as.AddChange(&Change{Action: Mount, Entry: entry})
+	}
+
+	currentAfter, err := applyProfile(ctx, currentBefore, desired, as)
+	if err != nil {
+		return err
+	}
+
+	return ctx.SaveCurrentProfile(currentAfter)
+}
+
+func applyUserFstab(ctx MountProfileUpdateContext) error {
+	desired, err := ctx.LoadDesiredProfile()
+	if err != nil {
+		return err
+	}
+	debugShowProfile(desired, "desired mount profile")
+
+	current, err := ctx.LoadCurrentProfile()
+	if err != nil {
+		return err
+	}
+	debugShowProfile(current, "current mount profile")
+
+	as := ctx.Assumptions()
+	_, err = applyProfile(ctx, current, desired, as)
+	return err
+}
+
+func applyProfile(ctx MountProfileUpdateContext, currentBefore, desired *osutil.MountProfile, as *Assumptions) (*osutil.MountProfile, error) {
```

Comment:
The first argument changed name from "up" to "ctx" - but as its not used in this function this has no consequences.

Reply:
yeah, this prepares everything for eventual rename to `upCtx` in the last patch in this series.